### ISSUE TITLE
🐛 Bug Report: Templating Secrets are shown as plain text in scaffolder

### ISSUE THREAD

**Issue by dariozachow** [backstage/backstage#23413]
### 📜 Description

When having an input field with the ui:widget: password the input field is covered, but the logs are showing the input values as plain text, when using the fetch:template action.
And everybody, who has access to the url with the id for the scaffolder task can open the logs and see those secrets.

### 👍 Expected behavior

All Secrets should be censored or not shown in the logs, which are available for everyone.

### 👎 Actual Behavior with Screenshots

![Screenshot 2024-03-06 at 11 35 09](https://github.com/backstage/backstage/assets/147033205/aac1e789-5eaf-490c-aedc-82168695cc89)

I can't really show the secrets but the token is shown in plain text.

### 👟 Reproduction steps

This Issue can be reproduced when having the following input, but works with every over input as well:

```yaml
    - title: Add Secrets
      properties:
        secret:
          title: my-secret
          type: string
          description: secret
          ui.autofocus: true
          ui:options:
            rows: 5
          ui:widget: password
```

And having the step fetch:template like this will print the secret:

```yaml
    - id: fetch-template
      name: Template starter repo
      action: fetch:template
      input:
        url: ./
        copyWithoutTemplating:
          - template.yaml
        values:
          secret: ${{ parameters.secret }}
   
```

### 🖥️ Your Environment

OS:   Darwin 23.3.0 - darwin/arm64
node: v21.6.0
yarn: 3.7.0
cli:  0.25.2 (installed)
backstage:  1.23.1

Dependencies:
  @backstage/app-defaults                                          1.5.0
  @backstage/backend-app-api                                       0.5.11
  @backstage/backend-common                                        0.19.9, 0.21.0
  @backstage/backend-defaults                                      0.2.10
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.3
  @backstage/backend-plugin-api                                    0.6.10
  @backstage/backend-tasks                                         0.5.15
  @backstage/backend-test-utils                                    0.3.0
  @backstage/catalog-client                                        1.6.0
  @backstage/catalog-model                                         1.4.4
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.3
  @backstage/cli                                                   0.25.2
  @backstage/config-loader                                         1.6.2
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.12.0
  @backstage/core-compat-api                                       0.2.0
  @backstage/core-components                                       0.13.10, 0.14.0
  @backstage/core-plugin-api                                       1.9.0
  @backstage/dev-utils                                             1.0.27
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.5
  @backstage/frontend-plugin-api                                   0.6.0
  @backstage/integration-aws-node                                  0.1.9
  @backstage/integration-react                                     1.1.24
  @backstage/integration                                           1.9.0
  @backstage/plugin-api-docs                                       0.11.0
  @backstage/plugin-app-backend                                    0.3.58
  @backstage/plugin-app-node                                       0.1.10
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.2
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.0
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.4
  @backstage/plugin-auth-backend-module-github-provider            0.1.7
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.7
  @backstage/plugin-auth-backend-module-google-provider            0.1.7
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.5
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.7
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.2
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.0
  @backstage/plugin-auth-backend-module-okta-provider              0.0.3
  @backstage/plugin-auth-backend                                   0.21.0
  @backstage/plugin-auth-node                                      0.4.4, 0.4.5
  @backstage/plugin-catalog-backend-module-gitlab                  0.3.7
  @backstage/plugin-catalog-backend-module-openapi                 0.1.27
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.7
  @backstage/plugin-catalog-backend-module-unprocessed             0.3.7
  @backstage/plugin-catalog-backend                                1.17.0
  @backstage/plugin-catalog-common                                 1.0.21
  @backstage/plugin-catalog-graph                                  0.4.0
  @backstage/plugin-catalog-import                                 0.10.6
  @backstage/plugin-catalog-node                                   1.7.0
  @backstage/plugin-catalog-react                                  1.10.0
  @backstage/plugin-catalog                                        1.17.0
  @backstage/plugin-events-backend-module-gitlab                   0.1.20
  @backstage/plugin-events-backend                                 0.2.19
  @backstage/plugin-events-node                                    0.2.19
  @backstage/plugin-explore-common                                 0.0.2
  @backstage/plugin-github-actions                                 0.6.11
  @backstage/plugin-home-react                                     0.1.8
  @backstage/plugin-home                                           0.6.2
  @backstage/plugin-kubernetes-backend                             0.15.0
  @backstage/plugin-kubernetes-common                              0.7.4
  @backstage/plugin-kubernetes-node                                0.1.4
  @backstage/plugin-kubernetes-react                               0.3.0
  @backstage/plugin-kubernetes                                     0.11.5
  @backstage/plugin-org                                            0.6.20
  @backstage/plugin-permission-backend                             0.5.33
  @backstage/plugin-permission-common                              0.7.12
  @backstage/plugin-permission-node                                0.7.21
  @backstage/plugin-permission-react                               0.4.20
  @backstage/plugin-proxy-backend                                  0.4.8
  @backstage/plugin-scaffolder-backend-module-azure                0.1.2
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.0
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.2
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.0
  @backstage/plugin-scaffolder-backend-module-github               0.2.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.13
  @backstage/plugin-scaffolder-backend                             1.21.0
  @backstage/plugin-scaffolder-common                              1.5.0
  @backstage/plugin-scaffolder-node                                0.3.0
  @backstage/plugin-scaffolder-react                               1.8.0
  @backstage/plugin-scaffolder                                     1.18.0
  @backstage/plugin-search-backend-module-catalog                  0.1.14
  @backstage/plugin-search-backend-module-explore                  0.1.14
  @backstage/plugin-search-backend-module-pg                       0.5.19
  @backstage/plugin-search-backend-module-techdocs                 0.1.14
  @backstage/plugin-search-backend-node                            1.2.14
  @backstage/plugin-search-backend                                 1.5.0
  @backstage/plugin-search-common                                  1.2.10
  @backstage/plugin-search-react                                   1.7.6
  @backstage/plugin-search                                         1.4.6
  @backstage/plugin-techdocs-backend                               1.9.3
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.5
  @backstage/plugin-techdocs-node                                  1.11.2
  @backstage/plugin-techdocs-react                                 1.1.16
  @backstage/plugin-techdocs                                       1.10.0
  @backstage/plugin-user-settings                                  0.8.1
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.6.0
  @backstage/test-utils                                            1.5.0
  @backstage/theme                                                 0.4.4, 0.5.1
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by awanlin** [comment]
Hi @dariozachow, have you looked at trying how it's documented here: https://backstage.io/docs/features/software-templates/writing-templates#using-secrets ? That uses `ui:field: Secret` and then pulls them from `${{ secrets.password }}` for example.

**Comment by dariozachow** [comment]
Hi @awanlin , thanks for your quick answer, but unfortunately this doesn't work.

I've updated my template the following way:

```yaml
  parameters:
    - title: Fill in some steps
      required:
        - secret
      properties:
        secret:
          title: Name
          type: string
          description: Unique name of the component
          ui:autofocus: true
          ui:field: Secret
          ui:widget: password
          ui:options:
            rows: 5

  steps:
    - id: fetch-template
      name: Template starter repo
      action: fetch:template
      input:
        url: ./
        copyWithoutTemplating:
          - template.yaml
        values:
          secret: ${{ secrets.secret }}       

```
But still the secret is displayed as plain text

![Screenshot 2024-03-06 at 12 02 30](https://github.com/backstage/backstage/assets/147033205/64385ef0-cf83-4824-94b6-3e82ab11dcaa)

**Comment by benjdlambert** [comment]
I've put together something over here: https://github.com/backstage/backstage/pull/23453

**Comment by dariozachow** [comment]
@benjdlambert Thanks for your work, creating this PR :)

**Comment by benjdlambert** [comment]
Closing as fixed in #23453
